### PR TITLE
Add task classes

### LINKED ISSUE #71: Add tasks
As a student, I want to be able to keep track of my projects' deadlines and tasks.

### PR DESCRIPTION
**Pull request by neyapraveen**
Add basic task classes for later implementation of task management
Fix #71

### CODE REVIEW

File: src/main/java/seedu/address/model/group/tasks/Task.java [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+import seedu.address.model.group.exceptions.TaskException;
+
+
+/**
+ * The Task class represents a task in the Duke chatbot application.
```
Comment (maj0-0) [inline src/main/java/seedu/address/model/group/tasks/Task.java:13]: Can change the Duke chatbot of javadocs

File: src/main/java/seedu/address/model/group/tasks/Deadline.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+     */
+    public String generateStr() {
+        return "D | " + (this.getStatus() == TaskStatus.DONE ? 1 : 0)
+                + " | " + this.getTask() + " | " + byStr;
+    }
+
+    /*/**
+     * Saves the `Deadline` task to a file if it has not been saved already.
+     */
+    /*@Override
```
Comment (wnchan) [inline src/main/java/seedu/address/model/group/tasks/Deadline.java:77]: this too

**Review by maj0-0 — COMMENTED** [review]
Aside from minor javadocs things, LGTM!

File: src/main/java/seedu/address/model/group/tasks/Event.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+     */
+    public String generateStr() {
+        return "E | " + (this.getStatus() == TaskStatus.DONE ? 1 : 0)
+                + " | " + this.getTask() + " | " + fromStr + " | " + toStr;
+    }
+
+    /*/**
+     * Saves the `Event` task to a file.
+     */
+    /*@Override
```
[the remark this answers is not in the document]
Comment (neyapraveen) [reply src/main/java/seedu/address/model/group/tasks/Event.java:87]: Yep, need to wait for the complete group implementation to see how to save. Saving of tasks isn't handled yet so I commented the code out for future use.

File: src/main/java/seedu/address/model/group/tasks/Task.java (hunk 1 above)
Comment (neyapraveen) [reply src/main/java/seedu/address/model/group/tasks/Task.java]: Changed, thank you!

File: src/main/java/seedu/address/model/group/tasks/Event.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (wnchan) [reply src/main/java/seedu/address/model/group/tasks/Event.java:87]: alright! LGTM!

**Pull request merged**